app/src/deploys/deploys.service.ts

```diff
@@ -1,16 +1,46 @@
-import { Injectable } from "@nestjs/common";
+import { CACHE_MANAGER, Inject, Injectable } from "@nestjs/common";
+import { Cache } from "cache-manager";
 import { CLValueParsers } from "casper-js-sdk";
+import { StatusCodes } from "http-status-codes";
 import { onChain } from "src/main";
+import { SidecarDeploy } from "src/types/api";
 import { DeployStatus, GetDeploy } from "src/types/deploy";
+import { ApiError } from "src/utils/ApiError";
 import {
   determineDeploySessionData,
   JsonDeploySession,
 } from "src/utils/deploy";
 
 @Injectable()
 export class DeploysService {
+  constructor(@Inject(CACHE_MANAGER) private readonly cacheManager: Cache) {}
+
   async getDeploy(hash: string): Promise<GetDeploy> {
-    const { deploy, executionResults } = await onChain.getDeploy(hash);
+    const cachedDeployByHash = await this.cacheManager.get<SidecarDeploy>(hash);
```

Comment:
What happens if this request fails for some reason?  Should this be wrapped in a try catch?

Reply:
Good question! This actually never fails in NestJS and only returns `null/undefined` (I forget which one) or whatever is fetched from the cache.

Reply:
Good to know!